nucleus/common/internal-api/src/main/java/fish/payara/internal/notification/PayaraNotification.java

```diff
@@ -52,60 +56,88 @@
     private String instanceName;
     private String subject;
     private String message;
+    private Serializable data;
+
+    private List<String> whitelist;
+    private List<String> blacklist;
 
     public String getEventType() {
         return eventType;
     }
 
-    public void setEventType(String eventType) {
+    protected void setEventType(String eventType) {
         this.eventType = eventType;
     }
 
     public String getServerName() {
         return serverName;
     }
 
-    public void setServerName(String serverName) {
+    protected void setServerName(String serverName) {
         this.serverName = serverName;
     }
 
     public String getHostName() {
         return hostName;
     }
 
-    public void setHostName(String hostName) {
+    protected void setHostName(String hostName) {
         this.hostName = hostName;
     }
 
     public String getDomainName() {
         return domainName;
     }
 
-    public void setDomainName(String domainName) {
+    protected void setDomainName(String domainName) {
         this.domainName = domainName;
     }
 
     public String getInstanceName() {
         return instanceName;
     }
 
-    public void setInstanceName(String instanceName) {
+    protected void setInstanceName(String instanceName) {
         this.instanceName = instanceName;
     }
 
     public String getSubject() {
         return subject;
     }
 
-    public void setSubject(String subject) {
+    protected void setSubject(String subject) {
         this.subject = subject;
     }
 
     public String getMessage() {
         return message;
     }
 
-    public void setMessage(String message) {
+    protected void setMessage(String message) {
         this.message = message;
     }
-}
\ No newline at end of file
+
+    public List<String> getWhitelist() {
+        return whitelist;
+    }
+
+    protected void setWhitelist(List<String> whitelist) {
+        this.whitelist = whitelist;
+    }
+
+    public List<String> getBlacklist() {
+        return blacklist;
+    }
+
+    protected void setBlacklist(List<String> blacklist) {
+        this.blacklist = blacklist;
+    }
+
+    public Serializable getData() {
+        return data;
+    }
+
+    protected void setData(Serializable data) {
+        this.data = data;
+    }
```

Comment:
It's immutable outside of the builder API with one notable caveat - if the notification is extended. I figured that since it has to be serializable that's not too much of an issue. The use case would be writing your own notification sender with extra field that your own receiver uses